### PR TITLE
add: signup,signin and signout + protected routes

### PR DESCRIPTION
**Pull request by thegeorgenikhil**
added
- signup functionality [Link](https://deploy-preview-1--gamesandbox.netlify.app/signup)
- signin functionality [Link](https://deploy-preview-1--gamesandbox.netlify.app/login)
- signout

Protected Routes
- user gets redirected to the login page if they try to access the wishlist or cart without logging in
[Try here - Cart Page](https://deploy-preview-1--gamesandbox.netlify.app/cart)
 Features not to be reviewed:
- remember me
- form validation

### CODE REVIEW

File: src/components/SignIn/SignIn.jsx [hunk 1]
```diff
@@ -34,7 +67,9 @@ const SignIn = () => {
             Forgot Password
           </Link>
         </div>
-        <button className="btn form-signup-btn">Login</button>
+        <button onClick={signInHandler} className={`btn form-signup-btn ${!isFormFullyFilled && "btn-disabled"}`}>
+          {isLoading ? <Loader /> : "Login"}
+        </button>
       </form>
       <p className="text-center">
         Don't have an account? Sign up!{" "}
```
Comment (OmKakatkar) [inline src/components/SignIn/SignIn.jsx]: Remove `{" "}` here

File: src/components/SignUp/SignUp.jsx [hunk 2]
```diff
@@ -1,43 +1,114 @@
...
+            type="checkbox"
+            name="hasUserAgreed"
+            onChange={(e) =>
+              setFormData({ ...formData, [e.target.name]: e.target.checked })
+            }
+          />
           <label htmlFor="">
             I have read and agreed to all terms and conditions
           </label>
         </div>
```
Comment (OmKakatkar) [inline src/components/SignUp/SignUp.jsx]: Add the value inside htmlFor as hasUserAgreed so that on clicking the label the checkbox will toggle

**Review by OmKakatkar — APPROVED** [review]
I liked:
* The use of PrivateRoute component. This was new for me.
* Meaningful variable names

Suggestion: 
Do look at the suggested changes in the comments

File: src/components/SignUp/SignUp.jsx (hunk 2 above)
Comment (thegeorgenikhil) [reply src/components/SignUp/SignUp.jsx]: added that, thank you

File: src/App.js [hunk 3]
```diff
@@ -17,8 +19,23 @@ function App() {
...
         <Route path="/forgot" element={<ForgotPassword />} />
         <Route path="/games/all" element={<AllGames />} />
-        <Route path="/cart" element={<Cart />} />
-        <Route path="/wishlist" element={<WishList />} />
+        <Route
+          path="/cart"
+          element={
+            <PrivateRoute>
+              <Cart />
+            </PrivateRoute>
```
Comment (Rohanmond) [inline src/App.js:27]: liked the use of private route

File: src/components/Navbar/Navbar.css [hunk 4]
```diff
@@ -54,20 +60,17 @@
...
 
-.nav-search-input{
+.nav-search-input {
   background-color: transparent;
   border: none;
   outline: none;
   font-size: 1rem;
   padding-left: 2px;
-  color: #FFFFFF;
+  color: #ffffff;
```
Comment (Rohanmond) [inline src/components/Navbar/Navbar.css]: better to use css variables

File: src/index.css [hunk 5]
```diff
@@ -191,7 +198,7 @@ body {
     gap: 20px;
   }
 
-  .wishlist-container{
+  .wishlist-container {
     display: flex;
     flex-wrap: wrap;
     gap: 20px;
```
Comment (Rohanmond) [inline src/index.css]: better to maintain same unit

**Review by Rohanmond — APPROVED** [review]

File: src/index.css (hunk 5 above)
Comment (thegeorgenikhil) [reply src/index.css]: made the required changes

**Pull request merged**